cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/RedisCache.java

```diff
@@ -267,17 +304,48 @@ public CacheData get(String type, String id, CacheFilter cacheFilter) {
         }
     }
 
+    private byte[] stringToBytes(String string) {
+        return string.getBytes(Charset.forName(HASH_CHARSET));
+    }
+
+    private byte[][] stringsToBytes(Collection<String> strings) {
+        final byte[][] results = new byte[strings.size()][];
+        int i = 0;
+        for (String string : strings) {
+          results[i++] = stringToBytes(string);
+        }
+        return results;
+    }
+
     private static class MergeOp {
         final Set<String> relNames;
         final List<String> keysToSet;
+        final Map<byte[], byte[]> hashesToSet;
+        final int skippedWrites;
 
-        public MergeOp(Set<String> relNames, List<String> keysToSet) {
+        public MergeOp(Set<String> relNames, List<String> keysToSet, Map<byte[], byte[]> hashesToSet, int skippedWrites) {
             this.relNames = relNames;
             this.keysToSet = keysToSet;
+            this.hashesToSet = hashesToSet;
+            this.skippedWrites = skippedWrites;
+        }
+    }
+
+    private boolean hashCheck(Map<String, byte[]> hashes, String id, String serializedValue, MessageDigest digest, List<String> keys, Map<byte[], byte[]> updatedHashes) {
```

Comment:
Javadoc comment on this method would be helpful for our future selfs (since it mutates a parameter).